app/src/org/odk/collect/android/widgets/IntentWidget.java

```diff
@@ -103,6 +92,39 @@ public void onClick(View v) {
         }
     }
 
+    public void makeButton(FormEntryPrompt prompt){
+        setOrientation(LinearLayout.VERTICAL);
+
+        TableLayout.LayoutParams params = new TableLayout.LayoutParams();
+        params.setMargins(7, 5, 7, 5);
+
+        // set button formatting
+        launchIntentButton = new Button(getContext());
+        setButtonLabel();
+        launchIntentButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, mAnswerFontsize);
+        launchIntentButton.setPadding(20, 20, 20, 20);
+        launchIntentButton.setEnabled(!prompt.isReadOnly());
+        launchIntentButton.setLayoutParams(params);
```

Comment:
If you merge with master (which you need to do anyways to make this auto-merge-able) you should use WidgetUtils.setupButton, which performs all this button setup code.